core/profile.c

```diff
@@ -849,7 +849,11 @@ static void calculate_ndl_tts(struct deco_state *ds, const struct dive *dive, st
 	const int ascent_s_per_deco_step = 1;
 	/* how long time steps in deco calculations? */
 	const int time_stepsize = 60;
-	const int deco_stepsize = 3000;
+	int deco_stepsize;
+	if (prefs.units.length == METERS)
+		deco_stepsize = 3000;
+	else
+		deco_stepsize = 3048;
```

Comment:
No.
https://github.com/subsurface/subsurface/blob/master/core/units.h#L26-L30

We've had this for ages and I am 99.99% sure that early on the deco code used this.
I'm too lazy to look for this, but yeah... this was a thing that I introduced a very long time ago.

Reply:
Yes, forgot about M_OR_FT. And yes, also forgot to check for not committing a step backwards in libdivecomputer. See, this time, I made sure the tests pass (as I expected this would affect the TestProfile but that seems to run only in sane unit mode). Hope this is better now.